### ISSUE TITLE
S3928 FP nameof(this.*)

### ISSUE THREAD

**Issue by Khaos66** [SonarSource/sonar-dotnet#2488]
### Description

I observed a false positive for rule 3928.

### Repro steps

Consider the following property:
```
   private string input;
   public string Request
   {
      get => this.input;
      set => this.input = value ?? throw new ArgumentNullException(nameof(this.Request));
   }
```

### Expected behavior

No warning

### Actual behavior

S3928 violation is reported by Sonar

### Known workarounds

* 

### Related information

* SonarC# Version 7.15 (build 8572)
* (Visual Studio Version 16.1.6)
* MS Build 15.9.21+g9802d43bc3
* Scanner for MSBuild Version: 4.6.2

**Comment by andrei-epure-sonarsource** [comment]
hi @Khaos66 . If will work if you pass `nameof(value)` - the actual parameter name is `value`, not `Request`.

**Comment by Corniel** [comment]
I also wonder, why do you think this is right behaviour?

**Comment by Khaos66** [comment]
Yes. My bad!
